### PR TITLE
i18n(de): Translate client-side-scripts.mdx

### PR DESCRIPTION
**Pull request by sarasate**
#### What kind of changes does this PR include?

- Translated content

#### Description

- Translate `client-side-scripts.mdx` to german

### CODE REVIEW

**Comment by Yan-Thomas** [comment]
Nice work @sarasate! This page was recently changed, could you please add the changes made? I'll review & merge soon after!
https://github.com/withastro/docs/commits/main/src/content/docs/en/guides/client-side-scripts.mdx?since=2022-12-21T15:02:53.000Z

**Comment by sarasate** [comment]
@Yan-Thomas Update done.

File: src/content/docs/de/guides/client-side-scripts.mdx [hunk 1]
```diff
@@ -0,0 +1,237 @@
+---
+title: Skripte und Ereignisbehandlung
+description: >-
+  Hinzufügen von clientseitiger Interaktivität zu Astro-Komponenten mit nativen Browser
+  JavaScript-APIs.
+i18nReady: true
+---
+
+Du kannst deinen Astro-Komponenten Interaktivität hinzufügen, ohne ein UI-Framework (/de/core-concepts/framework-components/) wie React, Svelte, Vue usw. zu verwenden, indem du standardmäßige HTML-`<script>`-Tags verwendest. So kannst du JavaScript im Browser ausführen lassen und deinen Astro-Komponenten Funktionen hinzufügen.
```
Comment (Yan-Thomas) [inline src/content/docs/de/guides/client-side-scripts.mdx:9]: There's a small mistake in the link syntax here:
```suggestion
Du kannst deinen Astro-Komponenten Interaktivität hinzufügen, [ohne ein UI-Framework](/de/core-concepts/framework-components/) wie React, Svelte, Vue usw. zu verwenden, indem du standardmäßige HTML-`<script>`-Tags verwendest. So kannst du JavaScript im Browser ausführen lassen und deinen Astro-Komponenten Funktionen hinzufügen.
```

File: src/content/docs/de/guides/client-side-scripts.mdx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+Standardmäßig verarbeitet und bündelt Astro `<script>`-Tags und unterstützt damit den Import von npm-Modulen, das Schreiben von TypeScript und mehr.
+
+## Verwendung von `<script>` in Astro
+
+In `.astro`-Dateien kannst du clientseitiges JavaScript hinzufügen, indem du ein (oder mehrere) `<script>`-Tags hinzufügst.
+
+In diesem Beispiel wird durch das Hinzufügen der Komponente `<Hallo />` zu einer Seite eine Nachricht auf der Browserkonsole aufgezeichnet.
+
+```astro title="src/components/Hello.astro"
```
Comment (Yan-Thomas) [inline src/content/docs/de/guides/client-side-scripts.mdx:41]: In the line above you translated the component name, but didn't change the code sample title. You can either keep the component name in English or change this, it's up to you.
````suggestion
```astro title="src/components/Hallo.astro"
````

File: src/content/docs/de/guides/client-side-scripts.mdx [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+- Wenn deine Komponente mehrmals auf einer Seite verwendet wird, wird das Skript nur einmal eingebunden.
+
+```astro title="src/components/Example.astro"
+<script>
+  // Verarbeitet! Gebündelt! TypeScript-unterstützt!
+  // Der Import von lokalen Skripten und Node-Modulen funktioniert.
+</script>
+```
+
+Um die Bündelung des Skripts zu vermeiden, kannst du die Direktive "is:inline" verwenden.
```
Comment (Yan-Thomas) [inline src/content/docs/de/guides/client-side-scripts.mdx:65]: Here `is:inline` should be wrapped in backticks.
```suggestion
Um die Bündelung des Skripts zu vermeiden, kannst du die Direktive `is:inline` verwenden.
```

File: src/content/docs/de/guides/client-side-scripts.mdx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+### Übergabe von Frontmatter-Variablen an Skripte
+
+In Astro-Komponenten läuft der Code in [Frontmatter](/de/core-concepts/astro-components/#das-komponentenskript) zwischen den `---`-Zeichen auf dem Server und ist im Browser nicht verfügbar. Um Variablen vom Server an den Client zu senden, brauchen wir eine Möglichkeit, unsere Variablen zu speichern und sie dann zu lesen, wenn JavaScript im Browser läuft.
+
+Eine Möglichkeit, dies zu tun, ist die Verwendung von [`data-*`-Attributen](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), um den Wert von Variablen in deiner HTML-Ausgabe zu speichern. Skripte, einschließlich benutzerdefinierter Elemente, können diese Attribute dann mit der Eigenschaft `dataset` eines Elements lesen, sobald dein HTML im Browser geladen ist.
+
+In dieser Beispielkomponente wird eine "Message"-Eigenschaft in einem "Data-Message"-Attribut gespeichert, sodass das benutzerdefinierte Element "this.dataset.message" lesen und den Wert der Eigenschaft im Browser abrufen kann.
```
Comment (Yan-Thomas) [inline src/content/docs/de/guides/client-side-scripts.mdx:190]: Similar issue, with words that should've been wrapped in backticks. Not sure if this is something particular to the German language, so feel free to ignore this if that's the case.
```suggestion
In dieser Beispielkomponente wird eine `message`-Eigenschaft in einem `data-message`-Attribut gespeichert, sodass das benutzerdefinierte Element `this.dataset.message` lesen und den Wert der Eigenschaft im Browser abrufen kann.
```

File: src/content/docs/de/guides/client-side-scripts.mdx [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+```astro title="src/components/AstroGreet.astro" {2} /data-message={.+}/ "this.dataset.message"
+---
+const { message = "Willkommen, Welt! } = Astro.props;
```
Comment (Yan-Thomas) [inline src/content/docs/de/guides/client-side-scripts.mdx:194]: There's a small code error here. This should fix the syntax highlighting for this line.
```suggestion
const { message = 'Willkommen, Welt!' } = Astro.props;
```

**Review by Yan-Thomas — COMMENTED** [review]
Thanks for the fast response @sarasate! I found a few issues, can you take a look at them? After this, LGTM.

**Comment by sarasate** [comment]
@Yan-Thomas Thanks for the review. Fixed your suggestions, seems to be fine now..

**Review by Yan-Thomas — APPROVED** [review]
LGTM!

**Review by Yan-Thomas — CHANGES_REQUESTED** [review]
Sorry, only noticed this now. We also need to add the `'guides/client-side-scripts'` key with the page sidebar name value to the [Deutsch `nav.ts` file](https://github.com/withastro/docs/blob/main/src/i18n/de/nav.ts) here, could you please add this? After this should be good to go, sorry for the back-and-forth 😅

**Comment by sarasate** [comment]
Ok, didn't know about that one, will keep that in mind. Updated.

**Review by Yan-Thomas — APPROVED** [review]

**Pull request merged**